app/scripts/views/mixins/settings-panel-mixin.js

```diff
@@ -10,14 +10,27 @@ define(function (require, exports, module) {
 
   var $ = require('jquery');
   var BaseView = require('views/base');
+  var KeyCodes = require('lib/key-codes');
 
   module.exports = {
     initialize: function (options) {
       this.parentView = options.parentView;
     },
     events: {
       'click .cancel': BaseView.preventDefaultThen('_closePanelReturnToSettings'),
-      'click .settings-unit-toggle': BaseView.preventDefaultThen('_triggerPanel')
+      'click .settings-unit-toggle': BaseView.preventDefaultThen('_triggerPanel'),
+      'keyup .settings-unit': 'onKeyUp'
+    },
+
+    onKeyUp: function (event) {
+      if (event.which === KeyCodes.ESCAPE) {
```

Comment:
I think this could be improved if `onKeyUp` immediately delegated to another function which does this check:

```js
onKeyUp: function (event) {
  this._hidePanelOnEscape(event.which);
},

_hidePanelOnEscape: function (keyCode) {
  if (keyCode == KeyCodes.ESCAPE) {
    this.hidePanel();
  }
}
```
Using this paradigm makes unit testing a bit simpler (because this branch isn't tested AFAICT). You can call `_hidePanelOnEscape` with a couple of key codes and ensure `hidePanel` is called when expected.